app/src/main/java/com/example/drawingactivity/DrawActivity.kt

```diff
@@ -89,88 +89,117 @@ class DrawActivity : AppCompatActivity() {
                     onEditText = { layerName -> viewModel.setCurrentText(layerName) },
                     onSubmit = { layerName ->
                         if (layerName != null) {
-                            viewModel.addLayer(layerName)
+                            viewModel.addToList(layerName)
                         } else viewModel.dialogCompleted()
                     },
                 )
-            } else if (!it.isDialogOpen) {
-                isAddDialogOpen = false
-            }
-            if (it.isLayerSheetOpen && !isLayerListViewSheetOpen) {
+            } else isAddDialogOpen = false
+            if (it.isLayerListViewSheetOpen && !isLayerListViewSheetOpen) {
                 isLayerListViewSheetOpen = true
-                viewModel.setLayerListViewState()
-                bottomSheetLambda = dialogUtils.layerListBottomSheet(context = this,
+                onLayerListUpdate = dialogUtils.layerListBottomSheet(context = this,
                     layerList = it.layerList,
-                    onDialogComplete = { viewModel.dialogCompleted() },
-                    onDelete = { index ->
-                        viewModel.deleteLayer(index)
-                        if (!viewModel.isLayerListEmpty()) {
-                            bottomSheetLambda?.invoke(index, null, DialogOperation.DELETE)
-                        }
+                    onSheetComplete = { viewModel.dialogCompleted() },
+                    onDelete = { targetLayer ->
+                        isDeleteDialogOpen = true
+                        viewModel.setDeleteDialogState()
+                        viewModel.setTargetString(targetLayer)
+                        dialogUtils.confirmationDialog(context = this@DrawActivity,
+                            title = this@DrawActivity.getString(R.string.deleteLayerDialogTitle),
+                            message = this@DrawActivity.getString(
+                                R.string.deleteLayerDialogBody, targetLayer
+                            ),
+                            onConfirm = {
+                                if (!viewModel.isListSizeZero()) {
```

Comment:
Why not instead not give them the dialog to confirm if it's zero? That makes more sense right?

Reply:
Yes, it does make more sense to not show the dialog at all if the length of the list is zero, adjusting for that

Reply:
Resolved.